include/fmt/format.h

```diff
@@ -3395,6 +3395,43 @@ inline typename buffer_context<Char>::type::iterator format_to(
                               basic_format_args<context>(as));
 }
 
+namespace internal {
+
+template<typename... Ts>
+struct void_ { typedef void type; };
+
+template <typename T, typename Enable = void_<>::type>
+struct it_category { typedef int type; };
+
+template <typename T>
+struct it_category<T*> { typedef std::output_iterator_tag type; };
+
+template <typename T>
+struct it_category<const T*> { typedef int type; };
+
+template <typename T>
+struct it_category<T, typename void_<typename T::iterator_category>::type> {
```

Comment:
So this is basically a SFINAE detection of presence of `T::iterator_category`? A comment would be nice as well.

Reply:
Yeah, I've added a comment to make this clear.